tools/format.py

```diff
@@ -48,42 +51,78 @@
+    )
+
+    if result.returncode == 0:
+        # Split output into lines, each line is a file path
+        modified_files = result.stdout.decode().splitlines()
+        return modified_files
+    else:
+        # If the diff command failed, return an empty list
+        return []
+
+
+# Filter the modified files against the include patterns
+def filter_files(modified_files, include_patterns):
+    included_files = []
+    for file in modified_files:
+        for pattern in include_patterns:
+            if fnmatch(file, pattern):  # Use fnmatch correctly
+                included_files.append(file)
+                break  # No need to check further patterns once it's matched
+    return included_files
+
+
+# Original include patterns
+include_patterns = [
+    "*.cpp",
+    "*.hpp",
+    "*.h",
+    "*.c",
+    "*.py",
+    "*.sh",
+    "*.cmake",
+    "*.proto",
+    "CMakeLists.txt",
+    "**/CMakeLists.txt",
+    "Dockerfile",
+    "**/Dockerfile",
+]
+
+# Get the list of modified files
+modified_files = get_modified_files()
+
+# Filter modified files against the include patterns
+files_to_format = filter_files(modified_files, include_patterns)
+
 # TODO: Fix this
-# formatters["licence"] = {
-#     "format": [
-#         [
-#             "licenseheaders",
-#             "-t",
-#             os.path.join(b.project_dir, ".licence.tmpl"),
-#             "--years={added}",
-#             "--owner=NUbots",
-#             f"--projname=NUbots",
-#             "--projurl=https://github.com/NUbots/NUbots",
-#             "-f",
-#             "{path}",
-#         ]
-#     ],
-#     "include": [
-#         "*.cpp",
-#         "*.hpp",
-#         "*.h",
-#         "*.c",
-#         "*.py",
-#         "*.sh",
-#         "*.cmake",
-#         "*.proto",
-#         "CMakeLists.txt",
-#         "**/CMakeLists.txt",
-#         "Dockerfile",
-#         "**/Dockerfile",
-#     ],
-#     "exclude": [
-#         "shared/utility/motion/splines/*",
-#         "shared/utility/platform/models/nugus/nugus.proto",
-#         "nusight2/src/assets/robot-models/nugus/nugus.proto",
-#         "module/support/optimisation/NSGA2Optimiser/src/nsga2",
-#     ],  # TODO exclude files that are not ours
-# }
+formatters["licence"] = {
+    "format": [
+        [
+            "licenseheaders",
+            "-t",
+            os.path.join(b.project_dir, ".licence.tmpl"),
+            "--years={added}",
+            "--owner=NUbots",
+            f"--projname=NUbots",
+            "--projurl=https://github.com/NUbots/NUbots",
+            "-f",
+            "{path}",
+        ]
+    ],
+    "include": files_to_format,
+    "exclude": [
+        "shared/utility/motion/splines/*",
+        "shared/utility/platform/models/nugus/nugus.proto",
+        "nusight2/src/assets/robot-models/nugus/nugus.proto",
+        "module/support/optimisation/NSGA2Optimiser/src/nsga2",
+    ],  # TODO exclude files that are not ours
```

Comment:
```suggestion
    ],  # TODO exclude files that are not ours
```
Do this TODO if possible and simple enough. Can we do a naive exclusion check? For example, files that are not ours are not going to be in the `include_patterns`

Reply:
if it is done, remove it

Reply:
We are trying to figure out what it means by "files that are not ours". @ysims or @jpptm, any ideas here?

Reply:
Anything in the include files list that shouldn't be included, because they're not our files to license. eg the nsga2 algorithm files aren't ours, so we exclude them. Same with Bit-Bot's spline files, as they are Bit-Bots owned. It looks to me like it's already done, it might just be left over from when @TrentHouliston first wrote this and wouldn't have known what should be excluded.

Reply:
Correct, anything that we copy and paste from somewhere else into our code from somewhere else with another licence shouldn't be excluded